tests/system/__init__.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Helpers for system/integration tests"""
+import shlex
+import subprocess
+import sys
+from contextlib import contextmanager
+from os import environ, getcwd
+from pathlib import Path
+from shutil import which
+from traceback import print_exc
+
+from pkg_resources import parse_version
+
+from pytest_virtualenv import VirtualEnv
+
+PROJECT_DIR = environ.get(
+    'TOXINIDIR',
+    Path(__file__).resolve().parent.parent.parent)
+
+MIN_PIP_VERSION = parse_version('19.0.0')
+
+IS_WIN = sys.platform[:3].lower() == 'win'
+BIN = 'Scripts' if IS_WIN else 'bin'
+ENV = which('env') or '/usr/bin/env'
+# ^  For the sake of simplifying tests, we assume that even in Windows,
+#    env will be available (via msys/mingw)
+
+
+def is_venv():
+    """Check if the tests are running inside a venv"""
+    return hasattr(sys, 'real_prefix') or (sys.base_prefix != sys.prefix)
+
+
+@contextmanager
+def env(env_dict):
+    original = {k: environ[k] for k in env_dict.keys() if k in environ}
+    changed = [environ.putenv(k, v) or k for k, v in env_dict.items()]
+    try:
+        yield
+    finally:
+        # Restore original environment
+        for k in changed:
+            environ.unsetenv(k)
+        for k, v in original.items():
+            environ[k] = v
+
+
+def normalize_run_args(args):
+    """Make sure we have a flatten list of shell-words"""
+    if len(args) == 1:
+        if isinstance(args[0], str):
+            args = shlex.split(args[0], posix=not(IS_WIN))
+        elif isinstance(args[0], (list, tuple)):
+            args = args[0]
+
+    return list(args)
+
+
+def _insert_env(args):
+    """It seems that :mod:`subprocess` doesn't consider changes in the
+    environment variables after the main python process is started.
+    To workaround this we can always prepend the command with `env`, this
+    guarantees environment variables have up-to-date values.
+    """
+    if len(args) > 0 and args[0] not in ('env', ENV):
+        args.insert(0, ENV)
+
+    return args
+
+
+def run(*args, **kwargs):
```

Comment:
I wonder if we have this kind of functionality already in `pyscaffold.shell.ShellCommand` or if we could at least make use of it to avoid code duplication here.